### PR TITLE
Paginate kudosu ranking page

### PR DESCRIPTION
**Pull request by nanaya**
Up to 1000 instead of `MAX_RESULTS` (10k) because the query gets a bit slow. Also it seems rather silly with the end of the table only having single digit kudosu.

### CODE REVIEW

**Review by notbakaneko — APPROVED** [review]

**Comment by nanaya** [comment]
oh wait, the paginator didn't actually limit the page number

File: app/Http/Controllers/RankingController.php [hunk 1]
```diff
@@ -223,10 +223,15 @@ public function index($mode, $type)
...
+        static $maxResults = 1000;
+
+        $maxPage = $maxResults / static::PAGE_SIZE;
+        $page = min(get_int(request('page')) ?? 1, $maxPage);
+
         $scores = User::default()
             ->with('country')
             ->orderBy('osu_kudostotal', 'desc')
-            ->paginate(static::PAGE_SIZE, ['*'], 'page', null, 1000);
+            ->paginate(static::PAGE_SIZE, ['*'], 'page', $page, $maxResults);
```
Comment (notbakaneko) [inline app/Http/Controllers/RankingController.php:234]: actually, doesn't `paginate` only have 4 params? 🤔

**Review by notbakaneko — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/RankingController.php (hunk 1 above)
Comment (nanaya) [reply app/Http/Controllers/RankingController.php:234]: that's what you think 👀

Comment (notbakaneko) [reply app/Http/Controllers/RankingController.php:234]: oh it's a magic param :|

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**